### ISSUE TITLE
Allow data-entry updating utility functions?

### ISSUE THREAD

**Issue by jfy133** [sidora-tools/sidora.core#38]
I'm finally getting a little bit back into using pandora/sidora again, and I have just written a little function that allows you to add new tags to entries (which you could then use with `format_as_update_existing()` to do the whole uploading thing).

```r
add_new_tags <- function(old_tags, new_tags){
  old_tags <- str_split(old_tags, ",")
  new_tags <- str_split(new_tags, ",")
  
  append(old_tags, new_tags) %>% unlist %>% sort %>% unique %>% paste0(collapse = ",") %>% gsub("^,", "", .)
}
```

This brought up a question in my mind as whether we should provide 'modification' functions or not, and if yes, to what extent?

Thoughts @sidora-tools/core? 

Also @ivelsko would this be a sort of thing you would be interested in?

A more detailed example:

```r
library(dplyr)

add_new_tags <- function(old_tags, new_tags){
  old_tags <- str_split(old_tags, ",")
  new_tags <- str_split(new_tags, ",")
  
  append(old_tags, new_tags) %>% unlist %>% sort %>% unique %>% paste0(collapse = ",") %>% gsub("^,", "", .)
}


samples <- get_df("TAB_Sample", con)

# filter
samples_raw <- samples %>%
  filter(grepl("^ALA", sample.Full_Sample_Id)) %>%
  convert_all_ids_to_values() %>%
  filter(sample.Type_Group == "Calculus")

# update tag and convert for update exisiting entries format
samples_updated <- samples_raw %>%
  mutate(sample.Tags = map2(sample.Tags, "FoodTransforms,James Fellows Yates", add_new_tags)) %>%
  sidora.core::format_as_update_existing()

#write_csv(samples_updated, "Sample.csv")
```

**Comment by ivelsko** [comment]
Yes, I'm sure I would use this if it were available. Sometimes trying to use the Pandora upload sheets brings me close to tears. It would also avoid the awful reformatting by excel/libreOffice in cells that you don't want modified (like the Dairy Cultures notes and site names entries have lots of bizarre characters now).

I like the idea of being able to add/modify information, but I'm not comfortable with being able to remove data. I can see myself screwing something up and unintentionally deleting information, and I'd prefer not to be able to do that. Unless there's also a way to restore from an earlier version

**Comment by ivelsko** [comment]
Question though - will it be possible to do this on locked samples? I'd like that so I don't need to ask someone else with permission to make changes for me.

**Comment by jfy133** [comment]
No, you wouldn't be able to actually update Pandora via sidora, but would
rather give you all the tools to automate the production of an updated
sheet that you could then upload using Pandora's upload function.

**Comment by ivelsko** [comment]
Ah ok. Any way to make/modify an upload file without actually opening it (and potentially introducing formatting changes thereby) would be helpful
